mlir/lib/Dialect/MIOpen/Transforms/ConvToGemmLowering.cpp

```diff
@@ -177,6 +178,81 @@ void affixGridwiseGemmAttributes(Operation *convOp, Operation *gop,
   }
 }
 
+/// 0-initialize the output for a backward weight convolution which uses
+/// atomic adds.
+/// For f32 type, the output is the filter tensor.
+/// for f16 type, the output is the workspace.
+LogicalResult zeroInit(Conv2DBwdWeightOp op, PatternRewriter &b) {
+  auto loc = op.getLoc();
+  auto filterDataType =
+      op.filter().getType().cast<MemRefType>().getElementType();
+  Value output;
+  if (filterDataType == b.getF32Type()) {
+    output = op.filter();
+  } else if (filterDataType == b.getF16Type()) {
+    assert(op.workspace() && "Op has no workspace");
+    output = op.workspace();
+  } else {
+    llvm_unreachable("Incorrect memref type supplied");
+  }
+  auto outputDataType = output.getType().cast<MemRefType>().getElementType();
+  auto zeroOp = createZeroConstantOp(b, loc, outputDataType);
+  auto collapsedOutput = createCollapseShapeOp(b, loc, output);
+  ArrayRef<int64_t> collapsedOutputShape =
+      collapsedOutput.getType().cast<MemRefType>().getShape();
+  llvm::SmallVector<AffineForOp, 1> affineLoops;
+  for (auto bound : collapsedOutputShape) {
```

Comment:
`buildAffineLoopNest()` - this can be massively simplified

Reply:
Since we already collapsed the memref to 1D, we don't even need the logic to emit a nested loop here.